### DISCUSSION TITLE
Building Discussion

### DISCUSSION THREAD

**Discussion by Alex313031** [Alex313031/thorium-legacy#15] [General]
@gz83 Apparently I am blocked from messaging on facebook for 3 days. I have no idea what I did.

SO to continue from where we were on FB messenger, he made a new patch, can you try it out?

**Comment by gz83** [comment]
I will try the new patch and see again if it compiles properly.

@Alex313031

**Comment by gz83** [comment]
I've fixed all the errors in a couple of patches for supermium, and now that I've successfully compiled the browser and confirmed that it works on Win7, I'm going to go ahead and add other modifications to it.

@Alex313031

**Comment by gz83** [comment]
We need a total of three patches to make the new version work on Win7/8/8.1 systems.

However, I think the supermium patch may still have some files that need to be modified, as the patch introduces some unrelated changes, so I think the relevant patches still need to be further updated subsequently.

But anyway, I've managed to get the new version of Chromium working on Win7/8/8.1 systems, which is already a huge step forward, and I can bring these experiences to thorium browser as well.

@Alex313031 

https://drive.google.com/file/d/11Fh-Y-bfW4c7mAWpFxlxtpRdy7j-9VoS/view?usp=sharing
https://drive.google.com/file/d/12QcW_AtOIAa2T1jPvyMpl3z91QYjqJaa/view?usp=sharing
https://drive.google.com/file/d/1JuWV-_ydp1xQp_ZSOCeTxKq_Fa8Cv6li/view?usp=sharing

**Reply by Alex313031** [reply]
What unrelated Changes? I want to make sure we are not adding or removing anything from Supermium, to ensure stablility.

**Reply by gz83** [reply]
I've seen some Android related code in the supermium patch, I'll take the time to try to remove the useless code inside the patch, also I think the patch could probably continue to be improved, as I'm not sure supermium has fully restored the support code that was removed upstream

@Alex313031

**Comment by gz83** [comment]
Additionally, if we compile with 118.0.5991.0, we will also need to port three security patches from the main branch, as they fix the recent outbreak of vulnerabilities that have been exploited in the wild

https://drive.google.com/file/d/19yQ2Fxjsh3DotNwTaUCrYnHP4SbmwiRw/view?usp=sharing
https://drive.google.com/file/d/1SoKL6N5FbIMHjLZ1TWgPaH3NlwIRyqID/view?usp=sharing
https://drive.google.com/file/d/1cLL_orj-obFC3WvYFNLkWAugXOWDKCZj/view?usp=sharing

@Alex313031

**Reply by Alex313031** [reply]
Yeah, good work, we want to make sure it is as secure as possible.

**Comment by Alex313031** [comment]
@gz83 Also, while working on https://github.com/Alex313031/CR-87 I have made some changes to the source code and to depot_tools, which allow one to build Chromium on Windows 7/8/8.1. Up to version 101. 102 uses the 20348 SDK, which is incompatible with Win7.

The modifications change some header files in src, and in depot_tools it changes the python version to 3.9 (last version for win7), and changes update_depot_tools.bat and setup_color.py to work on win7. Version 85 - 101 uses the 19041 SDK and MSVS 2017 or 2019.

102 uses the 20348 SDK (incompatible) and MSVS 2022 (incompatible)

I did all this for building the last tag of M87 on my Win7 haswell workstation. 87 was the last to support adobe flash and ftp:// URLs. I still use it on linux and windows occasionally when I come across a website that uses either of those things. Otherwise, it is insecure to use because of its age. But otherwise, it supports most HTML5 standards and works with most websites.

I also built M69 from https://github.com/Alex313031/CR-69-72 on the same machine. M69 is interesting because it still has the old tabstrip, but is new enough to support widevine and new HTML5 CSS standards. It does break with some sites though. Reddit has styling messed up, and github doesn't show releases properly.

I would someday like to port M59 to Windows XP in my new winxp repo. This is because M49 is very out of date, and breaks on ~15 % of websites due to not supporting HTML5. M59 is the first with proper HTML5 support, and given the fact that M54 has been made to work on XP, it should not be *too* hard to get M59 to work.

**Reply by gz83** [reply]
thanks for your efforts

@Alex313031

**Comment by Alex313031** [comment]
@gz83 Thanks for your efforts.

Also, ironically while working on M87, I figured out an issue in Thorium that I have wanted to fix for a long time.

It is not a bug, but rather not really the desired behavior I want.

We have the patch to always show full URLs from ungoogled chromium, and then I added a patch to make the checkbox always be checked.
![Menu_001](https://github.com/Alex313031/thorium-win7/assets/45863095/0a7d3e1b-b489-4570-8313-7d5739cda14f)

However, what I really wanted is to always show full URLs *by default*, but to *still allow the user to change the setting*

Currently, it is forced to true in source code, and ignores the setting, there is no way to change it.

But I finally found where the initial preference default is set. Here: 
https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/ui/toolbar/chrome_location_bar_model_delegate.cc;l=292?q=kpreventurlelisionsinomnibox

We can then remove the two files related to the ungoogled patch and my checkbox patch, remove the link from PATCHES.md, and then just set that line to `true` instead of `false`. This will make it show full URLs by default, but the user can still disable it via the context menu.

I am going to make a PR with ungoogled Chromium, because they are all about user choice, and this patch removes the choice.
There is also another setting that I might disable. It allows extensions to change this setting. But that can be a risk, because a malicious extension could change it to hide the true origin of a URL. I want to make it so that nothing can change the setting except for manual intervention from the user.

**Reply by gz83** [reply]
OK, if you initiate a PR, please let me know

@Alex313031

**Comment by Ginibe** [comment]
Is it possible to use this version as a portable one? When you try to run a bat-file from a flash drive on another computer, all installed extensions disappear, except for the “built-in” ones.
Browser version - 109.0.5414.169 (Official build), (32 bit)
Environment: 
initial computer - Windows 7 SP1 Ultimate x64
another computer - Windows 7 SP1 Ultimate x86

Thank you!

**Reply by gz83** [reply]
Try to check the replies with newer dates in the following issues to see if you can make your browser user data portable.

https://github.com/ungoogled-software/ungoogled-chromium-windows/issues/36

**Reply by Ginibe** [reply]
Thanks

**Reply by Ginibe** [reply]
None of the proposed "crutches" helped make the browser portable.
I mean the following: the browser folder is located on an external drive, you need to periodically launch the browser on computers with different versions of Windows (win7 32bit, win7 64bit, win10 64bit) without losing add-ons, bookmarks and settings.
I suspect that all Chromium forks suffer from this without using a portable launcher, I tested Iridium and Cent. It's a pity, I really liked this fork.
